nupd/fetchers/github.py

```diff
@@ -33,18 +47,63 @@ class GHRepository:
     owner: str
     repo: str
     branch: str
-
+    meta: MetaInformation = field(
+        converter=lambda x: MetaInformation(**x)
+        if not isinstance(x, MetaInformation)
+        else x
+    )
     has_submodules: bool | None
     commit: Commit | None = field(
         converter=lambda x: Commit(**x)
         if not isinstance(x, Commit | None)
         else x
     )
-    meta: MetaInformation = field(
-        converter=lambda x: MetaInformation(**x)
-        if not isinstance(x, MetaInformation)
-        else x
-    )
+    latest_release: GitHubRelease | None
+
+    async def fetch_latest_release(
+        self, github_token: str | None = None
+    ) -> GitHubRelease | None:
+        """Fetch the latest release information for this repository."""
+        return await get_latest_release(
+            self.owner,
+            self.repo,
+            github_token=github_token,
+        )
+
+    async def fetch_tags(
+        self, github_token: str | None = None
+    ) -> c.Iterable[GitHubTag] | None:
+        """Fetch list of tags for this repository."""
+        return await get_tags(
+            self.owner,
+            self.repo,
+            github_token=github_token,
+        )
+
+    async def get_latest_version(
+        self, github_token: str | None = None
+    ) -> tuple[str, str] | None:
+        """Get the latest version from either releases or tags.
+
+        Returns a tuple of (version_type, version) where version_type is either 'release' or 'tag'.
+        Returns None if no releases or tags are found.
+        """
+        # First try to get the latest release
+        latest_release = await self.fetch_latest_release(
+            github_token=github_token
+        )
+        if latest_release is not None:
+            return ("release", latest_release.tag_name)
+
+        # If no releases, try to get the latest tag
+        tags = await self.fetch_tags(github_token=github_token)
+        if tags is not None and len(list(tags)) > 0:
+            latest_tag = next(
+                iter(tags)
+            )  # GitHub returns tags in descending order
+            return ("tag", latest_tag.name)
+
```

Comment:
If we use `yield` in `self.fetch_tags`, we can refactor this to serialize only the first found tag instead of all of them.

```suggestion
        tags = await self.fetch_tags(github_token=github_token)
        if tags is not None:
            try:
                # GitHub returns tags in descending order
                latest_tag = next(iter(tags))
                return ("tag", latest_tag.name)  # noqa: TRY300
            except StopIteration:
                pass
```